### PR TITLE
 New worker class that does all the work

### PR DESCRIPTION
**Pull request by annando**
The worker functionality now moved from ```include/poller.php``` and ```boot.php``` to ```src/Core/worker.php```

### CODE REVIEW

**Comment by MrPetovan** [comment]
Calling @rabuzarus and @tobiasd to the rescue, for another review with a fresh set of eyes.

**Comment by annando** [comment]
@MrPetovan yeah, it's late there over the atlantic, isn't it?

**Comment by MrPetovan** [comment]
No, I’m 6 hours before your time, but I’m on my phone, with a small screen that’s not ideal to review large PRs.

File: src/Core/Worker.php [hunk 1]
```diff
@@ -0,0 +1,993 @@
...
+use Friendica\Core\Worker;
+use Friendica\Util\Lock;
+
+use dba;
+use dbm;
+
+/**
+ * @file src/Core/Worker.php
+ *
+ * @brief Contains the class for all worker relevant stuff
```
Comment (tobiasd) [inline src/Core/Worker.php]: could you add a hint for what the workers are about? Could just be "for all worker (background process) relevant stuff" or so.

**Comment by tobiasd** [comment]
Looks ok, but wait for @fabrixxm or @rabuzarus for some 3rd or 4th ok ;-)

I guess you have checked for typos with the script in utils and you have it running on your production node?

**Comment by annando** [comment]
I always run experimental code on my production machine.

File: src/Core/Worker.php (hunk 1 above)
Comment (annando) [reply src/Core/Worker.php]: Done

File: src/Core/Worker.php [hunk 2]
```diff
@@ -0,0 +1,993 @@
...
+
+	public static function spawnWorker() {
+		$args = array("include/poller.php", "no_cron");
+		get_app()->proc_run($args);
+	}
+
+	/**
+	 * @brief Adds tasks to the worker queue
+	 *
+	 * @param (integer|array) priority or parameter array, $cmd atrings are deprecated and are ignored
```
Comment (fabrixxm) [inline src/Core/Worker.php]: > $cmd atrings

?

Comment (annando) [reply src/Core/Worker.php]: Yeah. You don't know the great "atrings"? ;-)

**Pull request merged**